tests/query/processors/test_mandatory_condition_applier.py

```diff
@@ -0,0 +1,134 @@
+import copy
+from typing import List
+
+import pytest
+
+from snuba.datasets.schemas import MandatoryCondition
+from snuba.datasets.schemas.tables import TableSource
+from snuba.query.conditions import (
+    OPERATOR_TO_FUNCTION,
+    BooleanFunctions,
+    ConditionFunctions,
+    binary_condition,
+    combine_and_conditions,
+)
+from snuba.query.expressions import Column, Literal
+from snuba.query.logical import Query
+from snuba.query.processors.mandatory_condition_applier import MandatoryConditionApplier
+from snuba.request.request_settings import HTTPRequestSettings
+
+test_data = [
+    (
+        "table1",
+        [
+            MandatoryCondition(
+                ["deleted", "=", "0"],
+                binary_condition(
+                    None,
+                    ConditionFunctions.EQ,
+                    Column("deleted", None, "deleted"),
+                    Literal(None, "0"),
+                ),
+            )
+        ],
+    ),
+    (
+        "table2",
+        [
+            MandatoryCondition(
+                ["time", "=", "1"],
+                binary_condition(
+                    None,
+                    ConditionFunctions.EQ,
+                    Column("time", None, "time"),
+                    Literal(None, "1"),
+                ),
+            ),
+            MandatoryCondition(
+                ["time2", "=", "2"],
+                binary_condition(
+                    None,
+                    ConditionFunctions.EQ,
+                    Column("time2", None, "time2"),
+                    Literal(None, "2"),
+                ),
+            ),
+        ],
+    ),
+]
+
+
+@pytest.mark.parametrize("table, mand_condition", test_data)
+def test_mand_condition(table: str, mand_condition: List[MandatoryCondition]) -> None:
+
+    body = {
+        "conditions": [
+            ["d", "=", "1"],
+            ["c", "=", "3"],
+            ["a", "=", "1"],
+            ["b", "=", "2"],
+        ],
+    }
+
+    body_copy = copy.deepcopy(body)
+
+    cols = None
+    consistent = True
+
+    query = Query(
+        body_copy,
+        TableSource(table, cols, mand_condition, ["c1"]),
+        None,
+        None,
+        binary_condition(
```

Comment:
Why don't you just like one or two conditions to your query instead of four?
It would make the test a lot less verbose and you would not loose any value from what you are testing.